### PR TITLE
Added Debug Print method to aid debugging

### PR DESCRIPTION
**Pull request by bhuvnesh2703**
Added Debug Print method to aid debugging

### CODE REVIEW

File: libgpopt/src/base/CDrvdProp.cpp [hunk 1]
```diff
@@ -14,24 +14,31 @@
...
-  //
-  //---------------------------------------------------------------------------
-  CDrvdProp::CDrvdProp()
-  {}
-
-  IOstream &operator << (IOstream &os, CDrvdProp &drvdprop)
-  {
-    return drvdprop.OsPrint(os);
-  }
+	//		ctor
```
Comment (hardikar) [inline libgpopt/src/base/CDrvdProp.cpp]: extra indentation; but do we need this comment all?

File: libgpopt/src/base/CDrvdProp.cpp [hunk 2]
```diff
@@ -14,24 +14,31 @@
...
+	CDrvdProp::CDrvdProp()
+	{}
+
+	IOstream &operator << (IOstream &os, CDrvdProp &drvdprop)
+	{
+	return drvdprop.OsPrint(os);
```
Comment (hardikar) [inline libgpopt/src/base/CDrvdProp.cpp]: missing indent

File: libgpopt/src/base/CDrvdPropCtxt.cpp [hunk 3]
```diff
@@ -13,12 +13,25 @@
...
+
 namespace gpopt {
 
-  IOstream &operator << (IOstream &os, CDrvdPropCtxt &drvdpropctxt)
-  {
-    return drvdpropctxt.OsPrint(os);
-  }
+	IOstream &operator << (IOstream &os, CDrvdPropCtxt &drvdpropctxt)
+	{
+	return drvdpropctxt.OsPrint(os);
```
Comment (hardikar) [inline libgpopt/src/base/CDrvdPropCtxt.cpp]: missing indent. (make sure you're using tabs)

File: libgpopt/src/base/CDrvdPropCtxt.cpp [hunk 4]
```diff
@@ -13,12 +13,25 @@
...
+	}
+
+#ifdef GPOS_DEBUG
+	void
+	CDrvdPropCtxt::DbgPrint() const
+	{
+		CAutoTrace at(m_pmp);
+		(void) this->OsPrint(at.Os());
```
Comment (hardikar) [inline libgpopt/src/base/CDrvdPropCtxt.cpp]: Is this the same as `at.Os() << *this;` ? For those without an overloaded `<<`, I presume.

File: libgpopt/include/gpopt/base/COrderSpec.h [hunk 5]
```diff
@@ -117,7 +117,12 @@ namespace gpopt
 					// print
 					IOstream &OsPrint(IOstream &os) const;
 
-					
+#ifdef GPOS_DEBUG
+					// debug print for interactive debugging sessions only
+					void DbgPrint() const;
+#endif // GPOS_DEBUG
+
+
```
Comment (dhanashreek89) [inline libgpopt/include/gpopt/base/COrderSpec.h]: Extra line

File: libgpopt/src/base/CEnfdProp.cpp [hunk 6]
```diff
@@ -13,12 +13,27 @@
...
+
 namespace gpopt {
 
-  IOstream &operator << (IOstream &os, CEnfdProp &efdprop)
-  {
-    return efdprop.OsPrint(os);
-  }
+	IOstream &operator << (IOstream &os, CEnfdProp &efdprop)
+	{
+	return efdprop.OsPrint(os);
```
Comment (hsyuan) [inline libgpopt/src/base/CEnfdProp.cpp]: indent

File: libgpopt/include/gpopt/base/CReqdProp.h [hunk 7]
```diff
@@ -96,6 +96,11 @@ namespace gpopt
 			virtual
 			IOstream &OsPrint(IOstream &os) const = 0;
 
+#ifdef GPOS_DEBUG
+			// debug print for interactive debugging sessions only
+			void DbgPrint();
```
Comment (hsyuan) [inline libgpopt/include/gpopt/base/CReqdProp.h]: some `void DbgPrint();` and some `void DbgPrint() const;`.
Better to make them consistent.

File: libgpopt/src/base/CDrvdPropCtxt.cpp (hunk 4 above)
Comment (bhuvnesh2703) [reply libgpopt/src/base/CDrvdPropCtxt.cpp]: yes

**Pull request merged**